webapp/channels/src/components/channel_select/channel_select.tsx

```diff
@@ -18,58 +18,70 @@ type Props = {
     selectDm: boolean;
 };
 
-export default class ChannelSelect extends React.PureComponent<Props> {
-    render(): JSX.Element {
-        const options = [
-            <option
-                key=''
-                value=''
-            >
-                {Utils.localizeMessage('channel_select.placeholder', '--- Select a channel ---')}
-            </option>,
-        ];
+const ChannelSelect = ({
+    channels,
+    selectOpen,
+    selectPrivate,
+    selectDm,
+    value,
+    onChange,
+}: Props) => {
+    const intl = useIntl();
 
-        this.props.channels.forEach((channel: Channel) => {
-            const channelName = channel.display_name || channel.name;
-            if (channel.type === Constants.OPEN_CHANNEL && this.props.selectOpen) {
-                options.push(
-                    <option
-                        key={channel.id}
-                        value={channel.id}
-                    >
-                        {channelName}
-                    </option>,
-                );
-            } else if (channel.type === Constants.PRIVATE_CHANNEL && this.props.selectPrivate) {
-                options.push(
-                    <option
-                        key={channel.id}
-                        value={channel.id}
-                    >
-                        {channelName}
-                    </option>,
-                );
-            } else if (channel.type === Constants.DM_CHANNEL && this.props.selectDm) {
-                options.push(
-                    <option
-                        key={channel.id}
-                        value={channel.id}
-                    >
-                        {channelName}
-                    </option>,
-                );
-            }
-        });
+    const options = [
```

Comment:
I wonder if it makes sense to use `useMemo` here. This is not really used as a prop to the child component, so there should be no re-render issues (in theory) but if we use a `useMemo` we may avoid iterating through all the channels on every re-render.

What do you think?

Reply:
Is that needed given the component is already wrapped in `React.memo`?

Reply:
In that case, I'm just going to merge this then since it's basically equivalent to the old behaviour